math/caffe/files/patch-cudnn.diff

```diff
@@ -1,11 +1,11 @@
---- Makefile.config.orig	2015-02-22 14:13:41.000000000 +0900
-+++ Makefile.config	2015-02-22 14:15:21.000000000 +0900
+--- Makefile.config.orig	2017-08-24 10:43:43.000000000 +0900
++++ Makefile.config	2017-08-24 10:43:54.000000000 +0900
 @@ -2,7 +2,7 @@
  # Contributions simplifying and improving our build system are welcome!
- 
+
  # cuDNN acceleration switch (uncomment to build with cuDNN).
 -# USE_CUDNN := 1
 +USE_CUDNN := 1
- 
```

Comment:
This space came from `diff -u`, why did you remove it?

Reply:
Before opening the PR, when I was upgrading the whole thing, I generated all the patches again with 
```diff -Naur original_file new_file > patch.txt```
So the removal of that spaces are from diff -Naur.

I've been looking to caffe source, and it seems that that lines doesn't contain any space (https://github.com/BVLC/caffe/blob/master/Makefile.config.example#L3 and https://github.com/BVLC/caffe/blob/master/Makefile.config.example#L6). So in fact, I think that it's the normal behaviour of diff, to just keep empty line.

And I think that the patch on master has does spaced which should not be. I've also checked against the sources that caffe port is pointing at, and neither on that lines there is no space: https://github.com/BVLC/caffe/blob/6d92d8fcfe0eea9495ffbc326256ec5b70c3eed1/Makefile.config.example#L6

So, I understand that no space should be on the diff file.

Reply:
> The lines common to both files begin with a space character in unified format.

I tried `diff -Naur` but these spaces indeed exist, did you use an editor that automatically removes trailing whitespaces?

Reply:
Uhm, damn.., seems that Atom removes trailing whitespaces.
Added all of them with vi, so now should follow diff spec.
Thanks for the link!